### PR TITLE
community[minor]: Update SingleStore vector store

### PR DESCRIPTION
**Pull request by volodymyr-memsql**
Update the `SingleStorevectorStore` class with hybrid search functionality.
This PR contains:
- Updated code (mostly, ported functionality that is already available in the python library)
- Test
- Updated document
- New example

### CODE REVIEW

**Comment by volodymyr-memsql** [comment]
Hello, @jacoblee93! There was an automated comment regarding env variables access from examples and tests, in my mind this is fine, as it is just an example of a possible way of connecting to the database.
Also, should I change anything in the PR before it can be merged?
Thank you!

File: libs/langchain-community/src/vectorstores/singlestore.ts [hunk 1]
```diff
@@ -212,10 +356,87 @@ export class SingleStoreVectorStore extends VectorStore {
...
+      whereArgs.push(query, filterThreshold.toString());
+      return `MATCH (${this.contentColumnName}) AGAINST (?) > ?`;
+    };
+    const filterByVectorClause = (): string => {
+      whereArgs.push(JSON.stringify(vector), filterThreshold.toString());
+      return this.distanceMetric === "DOT_PRODUCT"
+        ? `${this.distanceMetric}(${this.vectorColumnName}, JSON_ARRAY_PACK(?)) > ?`
+        : `${this.distanceMetric}(${this.vectorColumnName}, JSON_ARRAY_PACK(?)) < ?`;
+    };
+    const whereClouses: string[] = [];
```
Comment (jacoblee93) [inline libs/langchain-community/src/vectorstores/singlestore.ts:369]: nit: Spelling, should be `whereClauses`

**Comment by jacoblee93** [comment]
Hey @volodymyr-memsql - I'm getting `src/indexes/vector_stores/singlestore_hybrid_search.ts(55,5): error TS2345: Argument of type '{ searchStrategy: "FILTER_BY_TEXT"; filterThershold: number; }' is not assignable to parameter of type 'SearchConfig'` in one of the examples. Think you could have a look?

**Comment by volodymyr-memsql** [comment]
@jacoblee93, thanks for checking, I fixed the example code.

**Comment by volodymyr-memsql** [comment]
@jacoblee93, are any other changes needed for the PR?

**Comment by jacoblee93** [comment]
Thank you!

**Pull request merged**